tests/cpp_api/multiple_dataloaders_test/multiple_dataloaders_test.cpp

```diff
@@ -0,0 +1,193 @@
+    std::cerr << "Going to process images\n";
+    std::cerr << "Remaining images " <<  rocalGetRemainingImages(handle) << "\n";
+    high_resolution_clock::time_point t1 = high_resolution_clock::now();
+    int index = 0;
+    bool first_run = true;
+    while (rocalGetRemainingImages(handle) >= inputBatchSize) {
+        index++;
+        if (rocalRun(handle) != 0)
+            break;
+
+        RocalTensorList output_tensor_list = rocalGetOutputTensors(handle);
+
+        std::vector<int> compression_params;
+        compression_params.push_back(IMWRITE_PNG_COMPRESSION);
+        compression_params.push_back(9);
+
+        for (unsigned idx = 0; idx < output_tensor_list->size(); idx++) {
+            auto output_tensor = output_tensor_list->at(idx);
+            int h = output_tensor->shape().at(1) * output_tensor->dims().at(0);
+            int w = output_tensor->shape().at(0);
+            if (first_run) {
+                mat_input.emplace_back(cv::Mat(h, w, cv_color_format));
+                mat_output.emplace_back(cv::Mat(h, w, cv_color_format));
+            }
+
+            output_tensor->copy_data(mat_input[idx].data, ROCAL_MEMCPY_HOST);
+            mat_input[idx].copyTo(mat_output[idx](cv::Rect(0, 0, w, h)));
+
+            std::string out_filename = std::string(outName) + ".png";  // in case the user specifies non png filename
+            if (display_all)
+                out_filename = std::string(outName) + std::to_string(index) + std::to_string(idx) + ".png";  // in case the user specifies non png filename
+
+            if (color_format == RocalImageColor::ROCAL_COLOR_RGB24) {
+                cv::cvtColor(mat_output[idx], mat_color, CV_RGB2BGR);
+                if (DISPLAY)
+                    cv::imshow("output", mat_output[idx]);
+                else
+                    cv::imwrite(out_filename, mat_color, compression_params);
+            } else {
+                if (DISPLAY)
+                    cv::imshow("output", mat_output[idx]);
+                else
+                    cv::imwrite(out_filename, mat_output[idx], compression_params);
+            }
+        }
+        first_run = false;
+    }
+
+    high_resolution_clock::time_point t2 = high_resolution_clock::now();
+    auto dur = duration_cast<microseconds>(t2 - t1).count();
+    auto rocal_timing = rocalGetTimingInfo(handle);
+    std::cout << "Load     time " << rocal_timing.load_time << std::endl;
+    std::cout << "Decode   time " << rocal_timing.decode_time << std::endl;
+    std::cout << "Process  time " << rocal_timing.process_time << std::endl;
+    std::cout << "Transfer time " << rocal_timing.transfer_time << std::endl;
+    std::cout << ">>>>> Total Elapsed Time " << dur / 1000000 << " sec " << dur % 1000000 << " us " << std::endl;
+    for (unsigned i = 0; i < mat_input.size(); i++) {
+        mat_input[i].release();
+        mat_output[i].release();
+    }
+    rocalRelease(handle);
+    if (!output)
```

Comment:
@SundarRajan28  this `output` check should be done after lines 111 & 114 when the graph is processed Why at end of file